crates/mysten-network/src/server.rs

```diff
@@ -63,13 +63,8 @@ impl<M: MetricsCallbackProvider> ServerBuilder<M> {
         self
     }
 
-    pub async fn bind(self, addr: &Multiaddr, tls_config: Option<ServerConfig>) -> Result<Server> {
-        let http_config = self
-            .config
-            .http_config()
-            // Temporarily continue allowing clients to connection without TLS even when the server
-            // is configured with a tls_config
-            .allow_insecure(true);
+    pub async fn bind(self, addr: &Multiaddr, tls_config: ServerConfig) -> Result<Server> {
+        let http_config = self.config.http_config().allow_insecure(false);
```

Comment:
After this lands then i'll go and rip out the "allow insecure" logic from sui-http